doc/source/microsoft/step-zero-azure-autoscale.rst

```diff
@@ -0,0 +1,313 @@
+
+   .. code-block:: bash
+
+     az provider register --namespace microsoft.insights
+
+   To check the status of the registration, run the following command:
+
+   .. code-block:: bash
+
+     az provider show -n microsoft.insights
+
+   Once the application has been registered, navigate to your active subscription on the `Portal <https://portal.azure.com/>`_.
+
+   Under "Resources", select the VMSS.
+   It should be named something like ``aks-nodepool1-<random-str>-vmss``.
+
+   .. image:: ../_static/images/azure/select_vmss.png
+      :align: center
+
+   From the left-hand menu, select "Scaling".
+   Click the blue "Enable autoscaling" button and an autogenerated form for a scale condition will appear.
+   We will add two new rules to this condition:
+
+   * Increase the instance count by 1 when the average CPU usage over 10 minutes is greater than 70%
+   * Decrease the instance count by 1 when the average CPU usage over 10 minutes is less than 5%
+
+   .. image:: ../_static/images/azure/scale_condition.png
+      :align: center
+
+   Make sure the "Scale based on metric" option is selected and click "+ Add new rule", another autogenerated form will appear.
+   This will be pre-filled with the required settings to fulfill our first rule, so save it by clicking "Update" and click "+ Add new rule" again.
+
+   .. image:: ../_static/images/azure/scale_out.png
+      :align: center
+
+   The second form needs to be edited for the second rule to descrease the instance count by 1 when the average CPU usage over 10 minutes is less than 5%.
+   Save this rule and then save the overall scale condition, the cluster will be updated automatically.
+
+   .. image:: ../_static/images/azure/scale_in.png
+      :align: center
+
+   .. note::
+
+     This form can also be used to change ``--node-count``/``--min-count``/``--max-count`` that was set in step 7 by using the "Instance limits" section of the scale condition ("Default", "Minimum" and "Maximum" respectively).
+
+     If you prefer to use the command line, you can run the following:
+
+       .. code-block:: bash
+
+         az aks update \
+           --name <CLUSTER-NAME> \
+           --resource-group <RESOURCE-GROUP> \
+           --update-cluster-autoscaler \
+           --min-count <DESIRED-MINIMUM-COUNT> \
+           --max-count <DESIRED-MAXIMUM-COUNT> \
+           --output table
+
+     **Both** ``--min-count`` and ``--max-count`` must be defined.
+
+
+.. note::
+
+   If you create the cluster using the Azure Portal you must enable RBAC.
+   RBAC is enabled by default when using the command line tools.
+
+   If you use the Azure interactive shell the included version of Helm (2.10) has a known bug so the Secure Helm step in the :ref:`setup-helm` section of this guide will not work. Either skip this step, or download the latest version of Helm.
```

Comment:
The required version of `helm` to install all the features of this chart requires version `2.11`. It is specifically the part where using the kubernetes pod priority things along with user placeholders, to be able to scale up ahead of time of user arrivals that requires it.

This is probably also enforced here when using helm to install it, if server side of helm `tiller` is less than 2.11 it does not comprehend how to handle the podPriority things. See: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/58ac5aecccb3bc38958bb4d66ca3a57fa49e2754/jupyterhub/Chart.yaml#L10

Reply:
This I copied from the non-autoscaling docs, so I'll update it there too!

Reply:
@sgibson91 I added that to the original doc last year when I had some free Azure credits https://github.com/jupyterhub/zero-to-jupyterhub-k8s/commit/3a94644b161b746370361fe175e588708bdad432
If the default helm has since been updated you could delete the paragraph.

Reply:
I've never actually tried with the interactive shell, I will experiment tomorrow 😄